### PR TITLE
Write papers for bounds and keywords, and deferred constants as kind and rank

### PR DESCRIPTION
**Pull request by everythingfunctional**

### CODE REVIEW

File: J3-Papers/constant_bounds_status.txt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+In the course of developing the syntax for the template feature, we
+encountered a couple of tricky aspects to handle with respect to the
+possible separation and/or duplication of specifications using the
+REQUIREMENT and REQUIRES feature. This paper discusses the issue with
+respect to the rank, shape and bounds of deferred constants,
+the possible solutions, and the rationale for the solution chosen by
+the generics subgroup.
+
+3. Deferred Constant Array Bounds
+=================================
```
Comment (tclune) [inline J3-Papers/constant_bounds_status.txt:20]:
```suggestion
2. Deferred Constant Array Bounds
=================================
```

File: J3-Papers/keyword_argument_status.txt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    END FUNCTION
+  END INTERFACE
+END REQUIREMENT
+
+REQUIREMENT R1(T, G, ...)
+  TYPE, DEFERRED :: T
+  INTERFACE
+    FUNCTION G(B)
+      TYPE(T), INTENT(IN) :: B
+      TYPE(T) :: F
```
Comment (arjenmarkusdeltaresnl) [inline J3-Papers/keyword_argument_status.txt:45]: Should this not be: TYPE(T) :: G

File: J3-Papers/keyword_argument_status.txt [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  TYPE, DEFERRED :: T
+  INTERFACE
+    FUNCTION F(A)
+      TYPE(T), INTENT(IN) :: A
+      TYPE(T) :: F
+    END FUNCTION
+  END INTERFACE
+END REQUIREMENT
+
+REQUIREMENT R1(T, G, ...)
```
Comment (arjenmarkusdeltaresnl) [inline J3-Papers/keyword_argument_status.txt:40]: REQUIREMENT R2

File: J3-Papers/keyword_argument_status.txt [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+Subgroup discussed a few different options to address this problem.
+
+1. The names of the dummy arguments must agree in all specifications
+   of a deferred procedure
+2. Keyword arguments are not allowed for deferred procedures
+3. Keyword arguments are only allowed for deferred procedures if the
+   names of the dummy arguments are the same in all of its
+   specifications
+4. The names the dummy arguments of a deferred procedure are not defined
```
Comment (arjenmarkusdeltaresnl) [inline J3-Papers/keyword_argument_status.txt:70]: The names of the dummy arguments

File: J3-Papers/kind_and_rank_constants.txt [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+3.1 Deferred Constants Cannot Be Used to Specify Rank or Kind
+-------------------------------------------------------------
+
+This solution eliminates entire classes of use cases. Subgroup thus
+quickly rejected this as a possible solution.
+
+3.2 Deferred Kinds or Ranks Are Not Checked When Processing Template
+--------------------------------------------------------------------
+
+This solution pokes a giant whole in the idea of "strong concepts" and
```
Comment (arjenmarkusdeltaresnl) [inline J3-Papers/kind_and_rank_constants.txt:66]: A giant hole ...

Comment (tclune) [reply J3-Papers/kind_and_rank_constants.txt:66]: Not just parts of the giant ... :-)

**Comment by everythingfunctional** [comment]
Thank you @arjenmarkusdeltaresnl for pointing out the mistakes, and @tclune for the clever joke :stuck_out_tongue_closed_eyes:

**Pull request merged**